### ISSUE TITLE
FR: Windows Support

### ISSUE THREAD

**Issue by BradKML** [koreader/koreader#7653]
**Does your feature request involve difficulty completing a task? Please describe.**
As viewed in https://github.com/koreader/koreader#installation currently Koreader only supports Linux based systems and E-Readers, and there is no standalone ebook reader out there for Windows

**Describe the solution you'd like**
Koreader with Windows support (assuming KoReader + Chrome PWA is impossible)

**Describe alternatives you've considered**
Calibre bundles too much editing software with its reader

**Comment by Frenzie** [comment]
Windows is actually adding support for effortlessly running the Linux app: https://devblogs.microsoft.com/commandline/the-initial-preview-of-gui-app-support-is-now-available-for-the-windows-subsystem-for-linux-2/

Besides that, here are some references for the types of _mostly_ fairly minor fixes that would be required to attain a working Windows build:
https://github.com/koreader/koreader-base/pull/225
https://github.com/koreader/koreader/issues/4512
https://github.com/koreader/koreader-base/pull/1071
https://github.com/koreader/koreader-base/pull/1072

**Comment by NiLuJe** [comment]
I'm fairly sure @chrox actually did build it for Windows, so, chances are a mingw build might be mostly functional.

**Comment by NiLuJe** [comment]
In any case, that's not something any of us is willing to spend time on, so, the whole affair would be a PRs welcome ordeal ;).

**Comment by Frenzie** [comment]
@NiLuJe A 2014 era build, sure... ;-)

**Comment by CRC-Mismatch** [comment]
> Windows is actually adding support for effortlessly running the Linux app

:smirk: from a user's standpoint, "effortlessly" only if you're a developer on Windows (10+) and already have WSL2 correctly set up... I'm not aware of how the WSL2 complete setup process is currently, but back when I migrated from WSL1 to it, it took me a lot of reading, following manuals, googling for solutions and forcing full Windows release updates before I had it fully working. Also, it seems to me that most "new and advanced" features are only being targeted for Windows 10 Insider and latest Windows 11 releases - I'd be surprised if my current 21H2 is able to leverage this feature. In conclusion, although this feature makes it possible for at least a small portion of Windows power-users to use KOReader, it still leaves out the majority of Windows users.

I'm interested in researching this matter further, so maybe in the next months (or years :confounded:) I'll be able to come back with a PR!

**Comment by Frenzie** [comment]
You run `wsl --install` and then the AppImage just works. It's no longer insider whatever. No fiddling with X servers necessary.

The feature was announced a very long time ago and then it took an absolute eternity to actually be released so I have no idea since when it's actually been that way.

**Comment by rayman3003** [comment]
Please make the KOreader more popular, by releasing a Windows version of it.

**Comment by Frenzie** [comment]
It's not quite that simple unfortunately. Windows is more different than Android in many ways. But on the good side the differences don't *feel* annoying to me, because they have better reasons. ^_^

**Comment by NiLuJe** [comment]
They also have something called a budget, which allows them to just shove off some of that very specific and low-level porting work (which oftens require *very* particular and intimate platform knowledge) to a porting studio.

We don't. We're a (small) bunch of hobbyists. (And we happen to not actually run Windows most of the time, or one we enjoy working with/on).

**Comment by poire-z** [comment]
> Please make the KOreader more popular

But we don't want to make it more popular ! :)
We want something that runs well on _our_ eInk devices.
The fact that it works on Android is a side effect (with some additional hard work from some of us) that it works on Linux as all these devices runs Linux.
(Imagine a time where 90% of our issues are Windows specific ones, like _why don't the cursor or our highlights style look similar to other Windows apps?_ !)

> by releasing a Windows version of it.

Microsoft would have to release a `Windows 12ee (eInk edition)` first :)

**Comment by rayman3003** [comment]
> By contrast, a game with a more customized engine that does an independent port will probably take at least a couple of months to port to a new platform.

OK. How many months u need to port the KOreader to Windows? 8 Months is enough for u? What about 1 year? ☺️

> and it doesn't tend to be a platform we enjoy working with/on

Too bad 🥺 Bcuz I'm enjoying playing games on it.

> The fact that it works on Android is a side effect 

But I have an e-ink tablet that runs Android too. U know? 😉

> Imagine a time where 90% of our issues are Windows specific ones

As a professional Windows user that already installed more than 5000 windows apps so far, I would say that it's not true. 😎

> But we don't want to make it more popular

OK. Thats the true reason that I was looking for 🤓

**Comment by Frenzie** [comment]
> OK. How many months u need to port the KOreader to Windows? 8 Months is enough for u? What about 1 year? relaxed

The working proof of concept stuff was about KOReader.

**Comment by Frenzie** [comment]
I don't believe so, but I couldn't say for sure.

**Comment by Svetloslav** [comment]
Any chance things have changed and we could potentially see a Windows version? Would be highly appreciated.

**Comment by Frenzie** [comment]
If anyone's worked on making it build on Windows they haven't shared any patches.

**Comment by Korb** [comment]
"Koreader on Windows" #920

**Comment by ramizpolic** [comment]
Hey folks, is this still part of the roadmap? I can lend a hand here to make it work

**Comment by chrox** [comment]
I occasionally receive email notifications from this thread, but this time I have some good news to share with everyone. I’ve recently started a new reading software project called Readest. One of Readest’s goals is to provide a consistent reading experience across platforms. It currently supports Windows, macOS, Linux, and the Web, with Android and iOS support coming soon. We also plan to support reading progress sync with Koreader devices.

If you’re interested, you can check it out here: https://github.com/readest/readest.

**Comment by ryanwwest** [comment]
@chrox Looks cool, I'll try it out. Are you planning to offer an e-ink friendly theme/static UI option for the Android app? 

Eventually syncing with KOReader annotations would be cool as well (and Zotero annotations!).

**Comment by chrox** [comment]
@ryanwwest Yes, definitely! Some newer Android e-ink devices might be able to run Readest, as it currently requires a recent WebView to render documents. However, once we release the Android version, we will test its performance on e-ink devices and then adapt the UI and theme accordingly for a better e-ink reading experience.

**Comment by shuvashish76** [comment]
> You run wsl --install and then the AppImage just works. It's no longer insider whatever. No fiddling with X servers necessary.

Or install [nix via WSL](https://nixos.org/download/#nix-install-windows), the largest package manager :D

> As a professional Windows user that already installed more than 5000 windows apps so far...

As _professional_ windows user it should be easy for you to get used to WSL ;)

**Comment by rayman3003** [comment]
> As _professional_ windows user it should be easy for you to get used to WSL ;)

It wont install. shows this error:

![Image](https://github.com/user-attachments/assets/a433d0c6-c8ac-41ef-9f39-4306c203a1f0)

**Comment by shuvashish76** [comment]
> WSL versions 0.67.6 and above has systemd support. Follow [Microsoft’s systemd guide](https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl) to configure it, **and then install Nix** using:

It's literally mentioned in the 1st line, you've to enable and run the systemd service.
btw you don't need to go nix path if your intention is to install KOReader only. It's for _power users_.
Better to use AppImage as @Frenzie mentioned or Ubuntu package system (e.g. [video guide](https://www.youtube.com/watch?v=vxTW22y8zV8))

**Comment by Frenzie** [comment]
> It wont install. shows this error:

Don't use Nix unless you want to for other reasons. Run `wsl --install` as stated above, see [here](https://learn.microsoft.com/en-us/windows/wsl/install) for all the steps. The default distro of Ubuntu 22.04 or 24.04 will work fine. Then you can install the Deb package which will probably even give you a shortcut in the start menu.
